src/neo4j_genai/retrievers/text2cypher.py

```diff
@@ -140,7 +138,9 @@ def get_search_results(
                 query=validated_data.query_text,
             )
         else:
-            prompt = self.custom_prompt
+            prompt = self.custom_prompt.format(
```

Comment:
If parameters are expected in the custom prompt, do you think we can use `Text2CypherTemplate`? We can instantiate such a template with a custom prompt as well. Something like:
```
prompt_template = Text2CypherTemplate(template=custom_prompt)
```

That being said, the prompt template doesn't support at the moment the checks you implemented (the required `{query}` placeholder), but it makes sense to enforce it here, and having it in the base `PromptTemplate` class would make that feature available for the other types of prompts we have (for RAG for instance). What do you think?

Reply:
I think that makes sense to use the Template classes and extend the checks to other prompts as well. Would you like me to make the necessary changes on this branch?

Reply:
If you are comfortable with it, you can go ahead and implement it. Feel free to reach out of you have any question.

Reply:
No problem, I'll make the changes

Reply:
I changed all Template classes to contain a `query_text` placeholder and validate upon construction. The `format` methods of each Template class also now take `query_text` as an arg, but allow for the old input arg as well. A warning will occur if using the old input arg. I refactored tests to adhere to the argument changes.